### DISCUSSION TITLE
Support for microsoft/Florence-2-large-ft vision model

### DISCUSSION THREAD

**Discussion by AshD** [microsoft/onnxruntime-genai#658] [Model support]
microsoft/Florence-2-large-ft works really well and it is much smaller than Phi-3 vision. 

Would it be possible to convert this model in Onnx format?

Thanks,
Ash

**Comment by tarzain** [comment]
+1! Would love to have this in ONNX

**Comment by balachandarsv** [comment]
+1 It would be great to have this in ONNX.

**Comment by aciddelgado** [comment]
Thanks all for the suggestion; we'll add this model to our roadmap and we'll discuss the priority internally. Thanks!

**Comment by AshD** [comment]
Xenova created an Onnx version of the Florence model
https://huggingface.co/onnx-community/Florence-2-base-ft/tree/main/onnx

**Reply by nogifeet** [reply]
Is it possible to get access to scripts/branch used to convert to onnx ?

I want to convert some of my fine-tuned checkpoints to onnx

**Reply by Youho99** [reply]
I believe you can create your own onnx now

https://github.com/microsoft/onnxruntime-genai/blob/main/examples/python/awq-quantized-model.md

**Reply by nogifeet** [reply]
@Youho99 Looks like florence2 is still not supported using AWQ. 

Traceback (most recent call last):
  File "/content/awq-quantized-model.py", line 140, in <module>
    main()
  File "/content/awq-quantized-model.py", line 117, in main
    quantize_model(args)
  File "/content/awq-quantized-model.py", line 53, in quantize_model
    model = AutoAWQForCausalLM.from_pretrained(
  File "/content/AutoAWQ/awq/models/auto.py", line 70, in from_pretrained
    model_type = check_and_get_model_type(
  File "/content/AutoAWQ/awq/models/auto.py", line 48, in check_and_get_model_type
    raise TypeError(f"{config.model_type} isn't supported yet.")
TypeError: florence2 isn't supported yet.

**Comment by timothycarambat** [comment]
@AshD Been looking for that, thank you!

**Comment by trincadev** [comment]
> Xenova created an Onnx version of the Florence model https://huggingface.co/onnx-community/Florence-2-base-ft/tree/main/onnx

Thank you. Any news for support on python? Xenova's onnx florence-2 model requires transformers.js

**Comment by natke** [comment]
Thank you for your interest in Florence-2 V. We are looking into it. Moving this issue into Discussions

**Comment by Youho99** [comment]
+1

**Comment by callmehanyu** [comment]
+1

**Comment by qingfengcss** [comment]
Could the author write a document about how to convert florence model to onnx format?

**Comment by Youho99** [comment]
Add same support request for new phi-3.5-vision 
Discussion here : https://github.com/microsoft/onnxruntime-genai/discussions/827

**Comment by happyme531** [comment]
I made an onnxruntime inference script for Florence2:
https://huggingface.co/happyme531/Florence-2-base-ft-ONNX-RKNN2

**Reply by AshD** [reply]
Do you have a C# version of this script?

**Comment by Rm1n90** [comment]
@natke Any update regarding Florence 2?

**Comment by ievgeniilarin** [comment]
Please provide an official onnx for the model 🙏

Using [this](https://github.com/curiosity-ai/florence2-sharp) in the meantime.
